dlf/plugins/toolbox/tools/imagedownload/class.tx_dlf_toolsImagedownload.php

```diff
@@ -0,0 +1,154 @@
+
+			}
+
+			$this->piVars['double'] = \TYPO3\CMS\Core\Utility\MathUtility::forceIntegerInRange($this->piVars['double'], 0, 1, 0);
+
+		}
+
+		// Load template file.
+		if (!empty($this->conf['toolTemplateFile'])) {
+
+			$this->template = $this->cObj->getSubpart($this->cObj->fileResource($this->conf['toolTemplateFile']), '###TEMPLATE###');
+
+		} else {
+
+			$this->template = $this->cObj->getSubpart($this->cObj->fileResource('EXT:dlf/plugins/toolbox/tools/imagedownload/template.tmpl'), '###TEMPLATE###');
+
+		}
+
+		// Get single page downloads.
+		$markerArray['###IMAGE_LEFT###'] =  $this->piVars['double'] == 1 ? $this->getImage($this->piVars['page'], $this->pi_getLL('leftPage', '')) : $this->getImage($this->piVars['page'], $this->pi_getLL('singlePage', ''));
+
+		// Get work download.
+		$markerArray['###IMAGE_RIGHT###'] = $this->piVars['double'] == 1 ? $this->getImage($this->piVars['page'] + 1, $this->pi_getLL('rightPage', '')) : '';
+
+		$content .= $this->cObj->substituteMarkerArray($this->template, $markerArray);
+
+		return $this->pi_wrapInBaseClass($content);
+
+	}
+
+
+	/**
+	 * Get image's URL and MIME type
+	 *
+	 * @access	protected
+	 *
+	 * @param	integer		$page: Page number
+	 * @param	string		$label: Link title and label
+	 *
+	 * @return	string	linkt to image file with given label
+	 */
+	protected function getImage($page, $label) {
+
+		$image = array ();
+
+		// Get @USE value of METS fileGrp.
+		$fileGrps = \TYPO3\CMS\Core\Utility\GeneralUtility::trimExplode(',', $this->conf['fileGrpsImageDownload']);
+
+		while ($fileGrp = @array_pop($fileGrps)) {
+
+			// Get image link.
+			if (!empty($this->doc->physicalStructureInfo[$this->doc->physicalStructure[$page]]['files'][$fileGrp])) {
+
+				$image['url'] = $this->doc->getFileLocation($this->doc->physicalStructureInfo[$this->doc->physicalStructure[$page]]['files'][$fileGrp]);
+
+				$image['mimetype'] = $this->doc->getFileMimeType($this->doc->physicalStructureInfo[$this->doc->physicalStructure[$page]]['files'][$fileGrp]);
+
+				switch ($image['mimetype']) {
+					case 'image/jpeg': 	$mimetypeLabel = '(JPG)';
+						break;
+					case 'image/tiff': 	$mimetypeLabel = '(TIFF)';
+							break;
+					default:	$mimetypeLabel = '';
+				}
+				$linkConf = array (
+					'parameter' => $image['url'],
+					'title' => $label . ' ' . $mimetypeLabel,
+					'additionalParams' => '',
+				);
+
+				$imageLink = $this->cObj->typoLink($label . ' ' . $mimetypeLabel, $linkConf);
+
+				break;
+
+			} else {
+
+				if (TYPO3_DLOG) {
+
+					\TYPO3\CMS\Core\Utility\GeneralUtility::devLog('[tx_dlf_toolsImagedownload->getImage('.$page.')] File not found in fileGrp "'.$fileGrp.'"', $this->extKey, SYSLOG_SEVERITY_WARNING);
+
+				}
+
+			}
+
+		}
+
+		return $imageLink;
+
+	}
+
+}
+
+if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dlf/plugins/toolbox/tools/pdf/class.tx_dlf_toolsImagedownload.php'])	{
```

Comment:
There is a typo in the path. It should be "plugins/toolbox/tools/**imagedownload**/" instead of "plugins/toolbox/tools/**pdf**/".